avalanche/models/dynamic_optimizers.py

```diff
@@ -118,9 +273,13 @@ def update_optimizer(optimizer, new_params, optimized_params, reset_state=False)
 
     # Add newly added parameters (i.e Multitask, PNN)
     # by default, add to param groups 0
+
+    param_structure = OptimizedParameterStructure(new_params, optimizer, verbose=True)
+
     for key in not_in_old:
         new_p = new_params[key]
-        optimizer.param_groups[0]["params"].append(new_p)
+        group = param_structure[key].single_group
+        optimizer.param_groups[group]["params"].append(new_p)
         optimized_params[key] = new_p
         optimizer.state[new_p] = {}
```

Comment:
This is the major change that I need to make to update_optimizer. And I only require to build the structure for that in order to get the new parameters group assignment. Initially I wanted to replace the optimized_param_id and new_params structure with it but I think that would be unnecessarily complex. As it is now I just create that structure to get the group assignments and then drop it.

For the two other functions of update_optimizer (update existing ids and remove parameters), the multiple parameters groups is already supported.